src/document.rs

```diff
@@ -297,6 +297,20 @@ impl Document {
         Ok(html.to_owned())
     }
 
+    /// Extracts references from markdown content.
+    pub fn extract_references(&self) -> String {
+        let mut trail = String::new();
+        let re = Regex::new(r"(?m:^ {0,3}\[[^\]]+\]:.+$)").unwrap();
+
+        if re.is_match(&self.content) {
+            for mat in re.find_iter(&self.content) {
+                trail.push_str(&self.content[mat.0..mat.1]);
+                trail.push('\n');
+            }
```

Comment:
we can also use [itertools::join](https://docs.rs/itertools/0.5.9/x86_64-unknown-linux-gnu/itertools/fn.join.html) here

this will require adding `itertools` to `Cago.{toml,lock}`
@johannhof any objections?